src/figuresClasses.ts

```diff
@@ -1,19 +1,73 @@
-export interface Figure {
+type Color = 'red'| 'green' | 'blue';
+type Shape = 'triangle'| 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public shape: 'triangle',
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('error');
+    }
 
-}
+    if (a + b >= c || a + c >= b || b + c >= a) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
 
-export class Circle {
+    const square = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
 
+    return +square.toFixed(2);
```

Comment:
I don’t know how critical it is if the tests passed, but Math.floor !== toFixed
`a method getArea that returns the area of the figure rounded down to hundredths.`